src/plugins/explore/public/application/context/editor_context/editor_context.tsx

```diff
@@ -173,6 +177,7 @@ export const useEditorContext = (): EditorContextValue => {
       setBottomEditorText,
       query,
       prompt,
+      editorMode,
```

Comment:
i'm conflicted about this change. I see why it could be convenient to have it here, as the consumers don't need to have a `const editorMode = useSelector(selectEditorMode)`, but `editorMode` is a redux state while the rest of the things here are the monaco editor stuff (it is true that they use the redux state to compute some things, but the output ultimately are things related to the monaco editors)

I'm afraid with this change that it can get confusing where `editorMode` is ultimately coming from. It seems more clear to me (albeit more wordy) for consumers to use `useEditorContext()` for things related to monaco editor, and `useSelector()` (aka Redux) to get redux state.


Happy to discuss this more in detail if you disagree